Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -293,10 +294,59 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove(wsName);
   }
 
+  /// Test the Nexus geometry loader from file
+  void testExecNexusLOKI() {
+    LoadInstrument loaderLOKI;
+    TS_ASSERT_THROWS_NOTHING(loaderLOKI.initialize());
+
+    // create a workspace with some sample data
+    wsName = "LoadInstrumentTestLOKI";
+    Workspace_sptr ws =
+        WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+    Workspace2D_sptr ws2D = boost::dynamic_pointer_cast<Workspace2D>(ws);
+
+    // put this workspace in the data service
+    TS_ASSERT_THROWS_NOTHING(AnalysisDataService::Instance().add(wsName, ws2D));
+
+    const std::string definitionFile = "LOKI_Definition.hdf5";
+    loaderLOKI.setPropertyValue("Filename", definitionFile);
+    loaderLOKI.setPropertyValue("Workspace", wsName);
+    loaderLOKI.setProperty("RewriteSpectraMap", OptionalBool(true));
+    inputFile = loaderLOKI.getPropertyValue("Filename");
+
+    std::string result;
+    TS_ASSERT_THROWS_NOTHING(result = loaderLOKI.getPropertyValue("Filename"));
+    TS_ASSERT(!result.compare(inputFile));
+
+    TS_ASSERT_THROWS_NOTHING(result = loaderLOKI.getPropertyValue("Workspace"));
+    TS_ASSERT(!result.compare(wsName));
+
+    TS_ASSERT_THROWS_NOTHING(loaderLOKI.execute());
```

Comment:
I tend to avoid using these `TS_ASSERT_THROWS_NOTHING` statements as they are a pain for debugging. Error moves beyond the point it was created. Infact I also set `Algorithm::setRethrow(true)` to ensure I get the original exception.